### PR TITLE
Delete Line Item

### LINKED ISSUE #79: [Budget Page] Delete Line Item

### PR DESCRIPTION
**Pull request by eliotmjohnson**

### CODE REVIEW

File: src/app/components/budget-category-card/budget-category-card.component.ts [hunk 1]
```diff
@@ -73,7 +74,30 @@ export class BudgetCategoryCardComponent implements AfterViewChecked {
         saveLineItemPayload.budgetCategoryId = this.budgetCategoryId;
 
         this.lineItemService.saveNewLineItem(saveLineItemPayload);
-        this.isAddingLineItem = false;
+        this.lineItemService.newlyAddedLineItemId
```
Comment (johnsonav1992) [inline src/app/components/budget-category-card/budget-category-card.component.ts:77]: Is this something that required rxjs specifically?

File: src/app/components/budget-category-item/budget-category-item.component.ts [hunk 2]
```diff
@@ -159,6 +162,12 @@ export class BudgetCategoryItemComponent implements OnInit, AfterViewChecked {
         }
     }
 
+    deleteLineItem() {
+        console.log(this.itemId);
```
Comment (johnsonav1992) [inline src/app/components/budget-category-item/budget-category-item.component.ts]: Need this?

File: src/app/services/line-item.service.ts [hunk 3]
```diff
@@ -36,4 +42,23 @@ export class LineItemService {
             }
         });
     }
+
+    deleteLineItem(lineItemId: string) {
+        this.http.delete(`${this.baseUrl}/${lineItemId}`).subscribe({
+            next: (res) => {
+                console.log(res);
```
Comment (johnsonav1992) [inline src/app/services/line-item.service.ts]: Need this one?

File: src/app/components/budget-category-card/budget-category-card.component.ts (hunk 1 above)
Comment (eliotmjohnson) [reply src/app/components/budget-category-card/budget-category-card.component.ts:77]: Yeah, in this scenario it was best.

File: src/app/components/budget-category-item/budget-category-item.component.ts (hunk 2 above)
Comment (eliotmjohnson) [reply src/app/components/budget-category-item/budget-category-item.component.ts]: Will fix

File: src/app/services/line-item.service.ts (hunk 3 above)
Comment (eliotmjohnson) [reply src/app/services/line-item.service.ts]: Will fix

File: src/app/components/budget-category-card/budget-category-card.component.ts (hunk 1 above)
Comment (eliotmjohnson) [reply src/app/components/budget-category-card/budget-category-card.component.ts:77]: it was an async operation hence had to use an observable

**Pull request merged**